services/app-web/src/common-code/data-helpers/makeDocumentKeyLookupList.ts

```diff
@@ -0,0 +1,52 @@
+import { base64ToDomain } from '../proto/stateSubmission'
+import { HealthPlanPackage } from '../../gen/gqlClient'
+import { parseKey } from '../s3URLEncoding'
+
+type LookupListType = {
+    currentDocuments: string[]
+    previousDocuments: string[]
+}
+
+const getKey = (s3URL: string) => {
+    const key = parseKey(s3URL)
+    return key instanceof Error ? null : key
+}
+
+export const makeDocumentList = (
+    submissions: HealthPlanPackage
+): LookupListType | undefined => {
+    const docBuckets = [
+        'contractDocuments',
+        'rateDocuments',
+        'documents',
+    ] as const
+    const lookupList: LookupListType = {
+        currentDocuments: [],
+        previousDocuments: [],
+    }
+    if (submissions) {
+        const revisions = submissions.revisions
+        if (revisions.length > 1) {
+            revisions.forEach((revision, index) => {
+                const revisionData = base64ToDomain(revision.node.formDataProto)
+                if (revisionData instanceof Error) {
+                    console.error(
+                        'failed to read submission data; unable to display document dates'
+                    )
+                    return
```

Comment:
There are a couple different options of what to do if we fail to unbox the revision proto. 
1. abort and return an error. If we want a user to see an error if any of the previous revisions don’t decode, this function should return an error directly so that calling code can decide how to surface that. 
2. Ignore the documents from that revision and continue, returning all the documents we are able to decode. This is a little attractive to me b/c as is we know we have data that can become invalid and we don’t really want to lose this whole feature just because an old revision can’t be read anymore, that said, it would mean that a user might get some funky unexpected file-deletion behavior, which seems bad

I think we should probably go with option 1 for now, and treat these errors as very unexpected. If that bites us we can always relax later.